stackdriver.go

```diff
@@ -71,19 +98,52 @@ type Options struct {
 	// Optional.
 	BundleCountThreshold int
 
-	// Resource is an optional field that represents the Stackdriver
-	// MonitoredResource, a resource that can be used for monitoring.
-	// If no custom ResourceDescriptor is set, a default MonitoredResource
-	// with type global and no resource labels will be used.
-	// Optional.
+	// Resource sets the MonitoredResource against which all views will be
+	// recorded by this exporter.
+	//
+	// All Stackdriver metrics created by this exporter are custom metrics,
+	// so only a limited number of MonitoredResource types are supported, see:
+	// https://cloud.google.com/monitoring/custom-metrics/creating-metrics#which-resource
+	//
+	// An important consideration when setting the Resource here is that
+	// Stackdriver Monitoring only allows a single writer per
+	// TimeSeries, see: https://cloud.google.com/monitoring/api/v3/metrics-details#intro-time-series
+	// A TimeSeries is uniquely defined by the metric type name
+	// (constructed from the view name and the MetricPrefix), the Resource field,
+	// and the set of label key/value pairs (in OpenCensus terminology: tag).
+	//
+	// If no custom Resource is set, a default MonitoredResource
+	// with type global and no resource labels will be used. If you explicitly
+	// set this field, you may also want to set custom DefaultMonitoringLabels.
+	//
+	// Optional, but encouraged.
 	Resource *monitoredrespb.MonitoredResource
 
-	// MetricPrefix overrides the OpenCensus prefix of a stackdriver metric.
-	// Optional.
+	// MetricPrefix overrides the prefix of a Stackdriver metric type names.
+	// Optional. If unset defaults to "OpenCensus".
 	MetricPrefix string
 
-	// DefaultTraceAttributes will be appended to every span that is exported.
+	// DefaultTraceAttributes will be appended to every span that is exported to
+	// Stackdriver Trace.
 	DefaultTraceAttributes map[string]interface{}
+
+	// DefaultMonitoringLabels are labels added to every metric created by this
+	// exporter in Stackdriver Monitoring.
+	//
+	// If unset, this defaults to a single label with key "opencensus_task" and
+	// value "go-<pid>@<hostname>". This default ensures that the set of labels
+	// together with the default Resource (global) are unique to this
+	// process, as required by Stackdriver Monitoring.
+	//
+	// If you set DefaultMonitoringLabels, make sure that the Resource field
+	// together with these labels is unique to the
+	// current process. This is to ensure that there is only a single writer to
+	// each TimeSeries in Stackdriver.
+	//
+	// Set this to &Labels{} (a pointer to an empty Labels) to avoid getting the
+	// default "opencensus_task" label. You should only do this if you know that
+	// the Resource you set uniquely identifies this Go process.
+	DefaultMonitoringLabels *Labels
```

Comment:
Suggestion: maybe also mention that unlike common labels, the label values of these default labels are fixed and that may potentially introduce high cardinality.

Reply:
What do you mean by common labels here? Why would being fixed introduce high cardinality - wouldn't it be the opposite (if they are not fixed)?

Reply:
For example if users are using pid as the DefaultMonitoringLabels, then each process will only have one label value for it, while other labels (e.g method, caller) can be broken down by different label values.